### PR TITLE
Collects user input and submits the new form data to the smart contract.

### PR DESCRIPTION
**Pull request by deweller**
Includes a basic form integration.

Does not include all form fields and does not have validation.

Implements #15

### CODE REVIEW

File: frontend/src/components/create-contract/PaymentSummary.tsx [hunk 1]
```diff
@@ -1,11 +1,29 @@
-const PaymentSummary = () => {
+import { CreateFormValues } from 'components/forms/types';
+
+//
+const PaymentSummary = ({
+  tokenName,
+  data,
+}: {
+  tokenName?: string;
+  data: CreateFormValues;
```
Comment (reisr3) [inline frontend/src/components/create-contract/PaymentSummary.tsx:9]: You can actually avoid the prop here!

Inside the body of this component, use `useFormikContext`. Destructure that into `values` (and lots of other stuff is available.)

Bonus points if we can wrap that in our own `useForm()` hook, to keep the components as readable as possible.

https://formik.org/docs/api/useFormikContext

File: frontend/src/components/create-contract/PaymentSummary.tsx [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+}) => {
+  const token = tokenName || process.env.REACT_APP_PAYMENT_TOKEN_NAME || '';
+
+  const bounty = data.bounty ? parseInt(data.bounty) : 0;
+  const buyIn = bounty * 0.1;
```
Comment (reisr3) [inline frontend/src/components/create-contract/PaymentSummary.tsx:14]: Assuming this is for testing, but we could just use the form value now right?

File: frontend/src/components/forms/types/CreateFormValues.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export type CreateFormValues = {
```
Comment (reisr3) [inline frontend/src/components/forms/types/CreateFormValues.d.ts:1]: Can we actually make this an interface and put it in the `interfaces` folder? `type` and `interface` are essentially the same thing in TS, but we're already using a ton of interfaces. Just a convention thing. Also don't need the `.d.ts`

Also though, probably don't need this at all once you add the Formik hook. If we type the form values using a generic, it should infer the types in the hook.

File: frontend/src/components/create-contract/CreateContractForm.tsx [hunk 4]
```diff
@@ -6,99 +6,197 @@ import { ListBox } from '../ListBox';
...
+    'acceptance-criteria': '',
+    bounty: '',
+  };
+
+  const history = useHistory();
+
+  const { contracts } = useSmartContracts();
+  const [submitting, setSubmitting] = useState(false);
+  const [formData, setValidatedFormData] =
+    useState<CreateFormValues>(initialValues);
```
Comment (reisr3) [inline frontend/src/components/create-contract/CreateContractForm.tsx:30]: Correct me if I'm wrong, but I don't think this is necessary with Formik. You can use their `validation` method to validate the form data, and then it should just be available as `values` in `onSubmit`

File: frontend/src/components/create-contract/CreateContractForm.tsx [hunk 5]
```diff
@@ -6,99 +6,197 @@ import { ListBox } from '../ListBox';
...
-          </div>
-          <div className="col-span-6 flex w-full bg-gray-50 px-4 py-5 sm:p-6">
-            <div className="w-1/2">
-              <PaymentSummary />
-            </div>
-            <div className="w-1/2">
-              <OptionsSummary />
+    <Formik
+      initialValues={initialValues}
+      validate={(values) => {
```
Comment (reisr3) [inline frontend/src/components/create-contract/CreateContractForm.tsx:52]: I'd just remove this validate prop for now, unless you want to move some of the validation logic in here. I think it's fine to do that in another PR though, to keep this one moving.

**Review by reisr3 — APPROVED** [review]
Looks great! Nothing that I'd hold up the PR for, but some suggestions. If you don't want to fix them now, feel free to just create some Issues and merge the PR in.

File: frontend/src/components/create-contract/PaymentSummary.tsx (hunk 2 above)
Comment (deweller) [reply frontend/src/components/create-contract/PaymentSummary.tsx:14]: I thought the value was actually hard-coded at 10%.

**Comment by deweller** [comment]
This is an incomplete implementation of the form.  I agree with all of the comments and added them as items for #16.
